src/main/java/seedu/address/model/appointment/AppointmentList.java

```diff
@@ -179,4 +179,22 @@ public void deleteAppointmentsWithNric(Nric nric) {
         requireNonNull(nric);
         internalList.removeIf(appointment -> appointment.getNric().equals(nric));
     }
+
+    /** Return true if new appt to be added overlaps with existing appointment of same Nric **/
+    public boolean samePatientHasOverlappingAppointment(Appointment targetAppt) {
+        requireNonNull(targetAppt);
+
+        for (Appointment appointment : this) {
+            // Check for same patient and same date
+            if (appointment.getNric().equals(targetAppt.getNric())
+                    && appointment.getDate().equals(targetAppt.getDate())) {
+                // Check for time overlap
+                if (appointment.hasOverlappingTimePeriod(targetAppt)) {
```

Comment:
Maybe we can try not to nest 2 if-loops here? Perhaps `hasOverlappingTimePeriod` can do these checks instead.

Reply:
Thanks! Will just return appointment.hasOverlappingTimePeriod(targetAppt) directly